clicommand/pipeline_upload.go

```diff
@@ -97,6 +98,11 @@ var PipelineUploadCommand = cli.Command{
 			Usage:  "Skip variable interpolation the pipeline when uploaded",
 			EnvVar: "BUILDKITE_PIPELINE_NO_INTERPOLATION",
 		},
+		cli.BoolFlag{
+			Name:   "reject-secrets",
+			Usage:  "When true, fail the pipeline upload early if the the pipeline contains secrets.",
+			EnvVar: "BUILDKITE_PIPELINE_REJECT_SECRETS",
```

Comment:
I wonder if this should be `BUILDKITE_AGENT_PIPELINE_REJECT_SECRETS` or `BUILDKITE_PIPELINE_UPLOAD_REJECT_SECRETS` 🤔

i.e. putting it in the `BUILDKITE_AGENT_` namespace, to separate it from things set by the Buildkite server, or making it clear that it only affects pipeline _uploads_.

I think there's a mixture of precedent. e.g:

- `BUILDKITE_PIPELINE_REPLACE` matches your new var name.
- `BUILDKITE_PIPELINE_NO_INTERPOLATION` also.
- `BUILDKITE_PIPELINE_UPLOAD_DRY_RUN` specifies `PIPELINE_UPLOAD` in the name.
- `BUILDKITE_AGENT_INCLUDE_RETRIED_JOBS` namespaces to `BUILDKITE_AGENT_`.

I think my preference would be `BUILDKITE_AGENT_PIPELINE_UPLOAD_REJECT_SECRETS` which is kinda too long but also very clear 🤷‍♂️

Here's some examples of server-specified build job environment vars with `BUILDKITE_PIPELINE_` namespace that I think we should avoid mixing these config flags up with:

```
BUILDKITE_PIPELINE_ID="350b5c5f-dafe-432b-a16a-4d730d3d81e1"
BUILDKITE_PIPELINE_NAME="Agent"
BUILDKITE_PIPELINE_PROVIDER="github"
BUILDKITE_PIPELINE_SLUG="agent"
```

Reply:
yeah, agreed, `BUILDKITE_AGENT_PIPELINE_UPLOAD_REJECT_SECRETS` is probably the least namespace-conflicty, and also does exactly what it says on the tin.

might be worth adding "standardise envar namespaces" to our list of things to do for v4.

Reply:
Looks like it's still `BUILDKITE_PIPELINE_UPLOAD_REJECT_SECRETS` in the code, which is fine if that's your decision, but it's different to your comment above 😉

Reply:
oh whoops! you're totally right, i can't read 😶

Reply:
fixed :)